frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/KafkaZKClient.java

```diff
@@ -110,22 +110,21 @@ public JSONObject getTopic(String topicName) throws Exception {
                 JSONObject broker = new JSONObject(new String(bytes, "UTF-8"));
                 String host = (String) broker.get("host");
                 Integer port = (Integer) broker.get("port");
-                endpoints.add(host + ":" + port);
+                endpoints.add(id + " : \"" + host + ":" + port + "\"");
             }
         } catch (Exception ex) {
             log.error("Failed to retrieve broker endpoints with exception: ", ex);
         }
         return endpoints;
     }
-
     public List<String> getBrokerDNSEndpoints(String frameworkName) {
         List<String> endpoints = new ArrayList<String>();
         try {
             List<String> ids = zkClient.getChildren().forPath(kafkaServicePath + BROKER_ID_PATH);
             for (String id : ids) {
                 byte[] bytes = zkClient.getData().forPath(kafkaServicePath + BROKER_ID_PATH + "/" + id);
                 JSONObject broker = new JSONObject(new String(bytes, "UTF-8"));
-                String host = "broker-" + id + "." + frameworkName + ".mesos";
+                String host = "kafka-" + id + "-broker." + frameworkName + ".mesos";
```

Comment:
Replace all this with:
`endpoints.add(String.format("kafka-%s-broker.%s.mesos:%s", id, frameworkName, broker.get("port")))`